### ISSUE TITLE
Shares: describe behaviour of slashes in query terms

### ISSUE THREAD

**Issue by JurgenR** [JurgenR/aioslsk#213]
Currently the query method of `SharesManager` operates on the entire path which allows query terms like "artist/song.mp3" to work. Backslashes are allowed in a path on Linux though, but perhaps it's easiest to just disregard any queries that have any form of slash in them

**Comment by JurgenR** [comment]
The entire query can be thrown away if it's in the inclusion and wildcard terms, in exclusion terms just the term can be thrown away

**Comment by JurgenR** [comment]
I made a mistake in my initial findings, since paths show up reversed in the client the query I built was wrong. Slashes do not get (entirely) ignored.

What I currently cannot verify is what happens on a different OS (MacOS), if instead of backslash a forward slash should be used to get matches. Furthermore I haven't checked if the alias is included.

For this ticket do the following:

* In `SharedItem.get_query_path`, return the normalized path instead of the local path. This way, backslashes can be used on all platforms. Unlike the official clients, the filename will be included (this is currently easier to implement)
* Document the behaviour in SOULSEEK.rst
* Write a test that includes backslashes

*Example*

If a user has the following files:

* File 1
** Filename: "the song.mp3"
** Directory: "band/cool/my" (reversed: "my/cool/band")
* File 2
** Filename: "the song.mp3"
** Directory: "band/great/my" (reversed: "my/great/band")

Queries:

* Using query "song -my\cool" (only as part of the directories) will only return result 2
* Using query "band\the song" will not return results, the filename does not seem to be included
